sources/packages/backend/libs/services/src/disbursement-overaward/disbursement-overaward.service.ts

```diff
@@ -76,6 +78,32 @@ export class DisbursementOverawardService {
     return result;
   }
 
+  /**
+   * Adds a legacy overaward value to disbursement overawards table.
+   * @param studentId id of the student.
+   * @param overawardValue overaward value.
+   * @param disbursementValueCode disbursement value code.
+   * @param entityManager entity manager used to perform the query.
+   */
+  async addLegacyOveraward(
+    studentId: number,
+    overawardValue: number,
+    disbursementValueCode: string,
+    entityManager?: EntityManager,
+  ): Promise<void> {
+    const repo =
+      entityManager?.getRepository(DisbursementOveraward) ??
+      this.disbursementOverawardRepo;
+    const auditUser = await this.systemUsersService.systemUser();
```

Comment:
The `auditUser` in this situation should probably be received as a parameter.
For instance, this method is used by the API to update the `LegacyOveraward` and the student user is the one to be saved as a `creator`.

Reply:
I believe that we are not on the same page on this one or I am missing something.
While saving the overawards I can see two scenarios:
1 - The Student is creating its account, then the student is the one causing the action, then the Student must be the user saved as `creator`.
2 - The overawards are imported from SFAS, and there is no user authenticated, then we consider this a system operation, then the `creator` will be defined as the `system-user`.
```ts
/**
   * Search for the student's SFAS data and update overaward values in
   * disbursement overawards table if any overaward values for BCSL or CSLF exist.
   * @param studentId student id for the disbursement overaward record.
   * @param lastName last name for sfas individuals.
   * @param sinNumber student's sin number.
   * @param entityManager entityManager to be used to perform the query.
   * @param auditUserId user that should be considered the one that is causing the changes.
   */
  async importSFASOverawards(
    studentId: number,
    lastName: string,
    birthDate: string,
    sinNumber: string,
    entityManager: EntityManager,
    auditUserId: number,
  ): Promise<void> {
    const sfasIndividual = await this.sfasIndividualService.getSFASOverawards(
      lastName,
      birthDate,
      sinNumber,
    );
    const overawards: DisbursementOveraward[] = [];
    if (sfasIndividual?.bcslOveraward > 0) {
      overawards.push({
        student: { id: studentId } as Student,
        disbursementValueCode: BC_STUDENT_LOAN_AWARD_CODE,
        overawardValue: sfasIndividual.bcslOveraward,
        creator: { id: auditUserId } as User,
      } as DisbursementOveraward);
    }
    if (sfasIndividual?.cslOveraward > 0) {
      overawards.push({
        student: { id: studentId } as Student,
        disbursementValueCode: CANADA_STUDENT_LOAN_FULL_TIME_AWARD_CODE,
        overawardValue: sfasIndividual.cslOveraward,
        creator: { id: auditUserId },
      } as DisbursementOveraward);
    }
    await this.disbursementOverawardService.addLegacyOverawards(
      overawards,
      entityManager,
    );
  }
```

Reply:
ok. Sorry. Got it now.